app/controllers/intents_controller.rb

```diff
@@ -1,9 +1,11 @@
 # frozen_string_literal: true
 
 class IntentsController < ApplicationController
-  def show
-    uri = Addressable::URI.parse(params[:uri])
 
+  before_action :check_uri
+  rescue_from Addressable::URI::InvalidURIError, with: :handle_invalid_uri
```

Comment:
Also there is another problem that if the uri is invalid, `Addressable::URI.parse` raises  `Addressable::URI::InvalidURIError`. So I added error handling for that error.